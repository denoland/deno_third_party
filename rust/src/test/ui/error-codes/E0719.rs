trait Foo: Iterator<Item = i32, Item = i32> {}
//~^ ERROR is already specified

type Unit = ();

fn test() -> Box<Iterator<Item = (), Item = Unit>> {
//~^ ERROR is already specified
    Box::new(None.into_iter())
}

fn main() {
    let _: &Iterator<Item = i32, Item = i32>;
    test();
}
