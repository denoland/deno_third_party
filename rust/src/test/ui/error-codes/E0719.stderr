error[E0719]: the value of the associated type `Item` (from the trait `std::iter::Iterator`) is already specified
  --> $DIR/E0719.rs:1:33
   |
LL | trait Foo: Iterator<Item = i32, Item = i32> {}
   |                     ----------  ^^^^^^^^^^ re-bound here
   |                     |
   |                     `Item` bound here first

error[E0719]: the value of the associated type `Item` (from the trait `std::iter::Iterator`) is already specified
  --> $DIR/E0719.rs:6:38
   |
LL | fn test() -> Box<Iterator<Item = (), Item = Unit>> {
   |                           ---------  ^^^^^^^^^^^ re-bound here
   |                           |
   |                           `Item` bound here first

error: aborting due to 2 previous errors

For more information about this error, try `rustc --explain E0719`.
